asapdiscovery-ml/asapdiscovery/ml/scripts/plot_data_efficiency.py

```diff
@@ -147,6 +148,104 @@ def losses_to_df(losses):
     return pandas.DataFrame({"split": all_labs, "loss": all_losses})
 
 
+def load_losses_dict(loss_dir, conv_function=None, take_best=True):
+    """
+    Load train, val, and test losses from `loss_dir`/loss_dict.json, converting as
+    necessary. Take loss from epoch with best original val loss.
+
+    Parameters
+    ----------
+    loss_dir : str
+        Directory containing pickle files
+    conv_function : callable, optional
+        If present, will use to convert mean absolute loss values
+
+    Returns
+    -------
+    pandas.DataFrame
+        DF with losses for the best idx across all splits
+    """
+    try:
+        loss_dict = json.load(open(f"{loss_dir}/loss_dict.json"))
+    except FileNotFoundError:
+        raise FileNotFoundError(f"No loss_dict.json file found in {loss_dir}")
+
+    # Load per-epoch losses for each molecule, and transpose so each row is an epoch and
+    #  each col is a molecule
+    all_losses = np.asarray(
+        [v["losses"] for d in loss_dict.values() for v in d.values()]
+    ).T
+    all_compound_ids = np.asarray(
+        [compound_id for d in loss_dict.values() for compound_id in d]
+    )
+    split_label_dict = {
+        compound_id: sp for sp, d in loss_dict.items() for compound_id in d
+    }
+
+    # Build and rearrange DF
+    df = pandas.DataFrame(all_losses, columns=all_compound_ids)
+    df = df.reset_index().melt(
+        id_vars=["index"], var_name="compound_id", value_name="loss"
+    )
+    df["split"] = [split_label_dict[c] for c in df["compound_id"]]
+    df = df.rename(columns={"index": "epoch"})
+
+    # Take only best idx
+    if take_best:
+        # Get best epoch idx from val set
+        val_arr = np.asarray([v["losses"] for v in loss_dict["val"].values()]).T
+        best_idx = np.argmin(val_arr.mean(axis=1))
+        df = df.loc[df["epoch"] == best_idx, :]
+
+    # Convert losses if desired
+    if conv_function:
+        df.loc[:, "loss"] = df["loss"].pow(0.5).apply(conv_function)
+
+    return df
+
+
+def load_all_losses_dict(in_df, rel_dir=None, conv_function=None, take_best=True):
```

Comment:
Perhaps consider the name collision between this and `def load_losses_dict`, these two were close enough to be a bit confusing.